service/collector.go

```diff
@@ -201,11 +201,21 @@ func (col *Collector) setupConfigurationComponents(ctx context.Context) error {
 	if err != nil {
 		return fmt.Errorf("failed to get config: %w", err)
 	}
+
+	// TODO: hacky workaround to unblock v0.50.0
+	configureGRPCLogging := false
+	if col.logger == nil {
+		configureGRPCLogging = true
+	}
 	if col.logger, err = telemetrylogs.NewLogger(cfg.Service.Telemetry.Logs, col.set.LoggingOptions); err != nil {
 		return fmt.Errorf("failed to get logger: %w", err)
 	}
 
-	telemetrylogs.SetColGRPCLogger(col.logger, cfg.Service.Telemetry.Logs.Level)
+	// setting the grpc logger multiple times causes
+	// a race condition
+	if configureGRPCLogging {
+		telemetrylogs.SetColGRPCLogger(col.logger, cfg.Service.Telemetry.Logs.Level)
+	}
```

Comment:
clean this logic and use a "sync.Once" for this behavior? Also this should be "global var sync.Once" since multiple collector instances should not race on this.

Reply:
Is it worth documenting that `cfg.Service.Telemetry.Logs.Level` is not reconfigurable when a ConfigProvider watches and returns a new config?

Reply:
this is only for grpc logs.

Reply:
Yes, I mean changing `cfg.Service.Telemetry.Logs.Level` will have no effect on grpc logs. Probably a minor detail.

Reply:
After reviewing this further, the problem is that the testbed is configuring the receivers before the service has a chance to configure the logger, using once won't help here as the grpc receivers are already starting before the grpc logger is set.

I was planning on adding an option to the collector to skip setting the grpc logger, which can then be configured in the InProcessCollector for the testbed. Thoughts?

Reply:
>I was planning on adding an option to the collector to skip setting the grpc logger, which can then be configured in the InProcessCollector for the testbed. Thoughts?

I think this is fine for the tests.